### PR TITLE
Dev/ft/ring 28492 update keyscheme

### PR DESCRIPTION
**Pull request by maximelubin**
Proper handling of erasure coding requires some change.
Also used several suggestions (having used code in hyperdrive keys for example).

### CODE REVIEW

File: Design.md [hunk 1]
```diff
@@ -34,23 +34,22 @@ Since we can specify which key to use on an hyperdrive, we could define a new ke
...
-<n_chunk> := number of splitted parts (1 for non-splitted objects)
+<placement_policy_version> := Natural (so 0 or 1 to start) - what placement policy was used
+<split> := <size>,<split_size>
+<size> := total size of the object
 <split_size> := size of each splitted parts, except last one (see hyperdrive keys below)
-<rep_policy> := RS,k,m or CP,n (for n-1 copies)
+<rep_policy> := RS,k,m,stripe_size or CP,n (for n-1 copies) - stripe_size is a positive integer
 <object_key> := parent S3 object key (or a prefix) - can contain anything but ‘#’
-<rand> := 64 bits random number (unicity inside 1 hyperdrive)
+<rand> := 32 bits random number (unicity inside 1 hyperdrive) encoded as hexadecimal string
```
Comment (xroche-scality) [inline Design.md:45]: Is this sufficient ? Collisions will ensue with 65K keys, or am I mistaken ?

**Review by xroche-scality — APPROVED** [review]

File: Design.md (hunk 1 above)
Comment (maximelubin) [reply Design.md:45]: Key is prefixed by object name. And as we discussed offline with Benoit, we need to add bucket as well (or a hash of both). Would be surprised to have 65K keys with same hash/name (effectively versions)

**Comment by maximelubin** [comment]
Not merging this right now. In designing the rest of the Zenko + hyperdrive set of functionalities, we (me, Xavier ad Benoit) have identified several other changes to do in the keyscheme (some of which must be confirmed with DB team).

**Comment by maximelubin** [comment]
Will do a new separate one instead. The split implementation relies on this one

**Pull request merged**